File: java/src/main/java/mx/iteso/test/YogurtIceCreamStoreTest.java
package mx.iteso.test;

import mx.iteso.Cone;
import mx.iteso.Stores.YogurtIceCreamStore;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

/**
 * Class for yogurt ice cream tests.
 */
public class YogurtIceCreamStoreTest {
    /**
     * testing Guava Ice Cream.
     */
    @Test
    public void testGuavaYogurtIceCream() {
        Cone cone = new YogurtIceCreamStore().prepareCone("yogurt", "Guava");
        assertEquals(cone.getIceCream().getFlavor(), "Guava yogurt ice cream");
    }

    /**
     * testing Mango Ice Cream.
     */
    @Test
    public void testMangoYogurtIceCream() {
        Cone cone = new YogurtIceCreamStore().prepareCone("yogurt", "Mango");
        assertEquals(cone.getIceCream().getFlavor(), "Mango yogurt ice cream");
    }

    /**
     * testing Strawberry Ice Cream.
     */
    @Test
    public void testStrawberryYogurtIceCream() {
    Cone cone = new YogurtIceCreamStore().prepareCone("yogurt", "Strawberry");
    assertEquals(cone.getIceCream().getFlavor(), "Strawberry yogurt ice cream");
    }

    /**
     * testing Pineapple Ice Cream.
     */
    @Test
    public void testPineappleYogurtIceCream() {
    Cone cone = new YogurtIceCreamStore().prepareCone("yogurt", "Pineapple");
    assertEquals(cone.getIceCream().getFlavor(), "Pineapple yogurt ice cream");
    }
}

File: csharp/factorypattern/test/stores/YogurtIceCreamStoreTest.cs
using NUnit.Framework;
using factorypattern.main;
using factorypattern.main.stores;

namespace factorypattern.test.stores
{
    [TestFixture]
    internal class YogurtIceCreamStoreTest
    {
        private YogurtIceCreamStore _yogurtIceCreamStore;
        private Cone _cono;

        [SetUp]
        public void setUp()
        {
            _yogurtIceCreamStore = new YogurtIceCreamStore();
            _cono = new Cone();
        }

        [TestCase]
        public void yogurtStoreTest()
        {
            _cono = _yogurtIceCreamStore.prepareCone("Mango");
            Assert.AreEqual("Mango Yogurt Ice Cream", _cono.getIceCream().getName());
            _cono = _yogurtIceCreamStore.prepareCone("Pineapple");
            Assert.AreEqual("Pineapple Yogurt Ice Cream", _cono.getIceCream().getName());
            _cono = _yogurtIceCreamStore.prepareCone("Strawberry");
            Assert.AreEqual("Strawberry Yogurt Ice Cream", _cono.getIceCream().getName());
            _cono = _yogurtIceCreamStore.prepareCone("Guava");
            Assert.AreEqual("Guava Yogurt Ice Cream", _cono.getIceCream().getName());

        }
    }
}
